### PR TITLE
ACMS-1903: Use local executable for drush and fallback to global.

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**

Fixes #ACMS-1903

**Proposed changes**
Use local executable for drush and fallback to global.

**Alternatives considered**
NA

**Testing steps**
Follow from ticket.

### CODE REVIEW

**Review by apathak18 — APPROVED** [review]

File: src/Helpers/Process/Commands/Drush.php [hunk 1]
```diff
@@ -20,6 +20,9 @@ public function getBaseCommand(): string {
   protected function getCommand(array $commands): array {
     $uri = $this->input->getParameterOption('--uri');
     $commands = parent::getCommand($commands);
+    if (!empty(shell_exec(sprintf("which %s", escapeshellarg('./vendor/bin/drush'))))) {
```
Comment (vishalkhode1) [inline src/Helpers/Process/Commands/Drush.php:23]: @rajeshreeputra Let's avoid this. This first looks for command and then update it. We can do following.
We can update `getCommand()` method of `CommandBase.php` like following:
is_executable($this->getBaseCommand())
```suggestion
  if (!is_executable($this->getBaseCommand())) {
    // Use ExecutableFinder to look for command globally.
  }
```


and then in Drush.php we can do following :
```suggestion
  public function getBaseCommand(): string {
    return './vendor/bin/drush';
  }
```

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

File: src/Helpers/Process/Commands/Drush.php (hunk 1 above)
Comment (rajeshreeputra) [reply src/Helpers/Process/Commands/Drush.php]: updated

**Review by vishalkhode1 — APPROVED** [review]

File: src/Helpers/Process/Commands/CommandBase.php [hunk 2]
```diff
@@ -167,22 +167,25 @@ public function getBaseCommand(): string {
...
-    }
+    $this->command = $this->getBaseCommand();
+    if (!is_executable($this->command)) {
+      preg_match("/[^\/]+$/", $this->command, $baseCommand);
+      $this->command = $baseCommand ? $baseCommand[0] : '';
+      // Use symfony executable finder object to look into
+      // global and application command to execute.
+      $executableFinder = new ExecutableFinder();
+      $this->command = $executableFinder->find($this->command, NULL, [
+        $this->rootDir . '/vendor/bin',
```
Comment (chandan-singh7929) [inline src/Helpers/Process/Commands/CommandBase.php:178]: I don't think this change is needed since you have already override in  `getBaseCommand()`

Comment (vishalkhode1) [reply src/Helpers/Process/Commands/CommandBase.php:178]: @chandan-singh7929 This is needed for extra directory to search for. For ex: in composer, the method `getBaseCommand()` returns `composer`. So, ExecutableFinder() class search if `composer` command is present in globally i.e in `/usr/local/bin` path. If it doesn't found then we've added as fallback that find this command in your project's `/vendor/bin` directory. If the same doesn't exist in global.

**Pull request merged**